conformance/utils/suite/suite.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package suite
+
+import (
+	"testing"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+)
+
+// ConformanceTestSuite defines the test suite used to run Gateway API
+// conformance tests.
+type ConformanceTestSuite struct {
+	Client           client.Client
+	RoundTripper     roundtripper.RoundTripper
+	GatewayClassName string
+	ControllerName   string
+	Debug            bool
+	Cleanup          bool
+}
+
+// New returns a new ConformanceTestSuite.
+func New(c client.Client, gcName string, debug, cleanup bool, roundTripper roundtripper.RoundTripper) *ConformanceTestSuite {
+	if roundTripper == nil {
+		roundTripper = &roundtripper.DefaultRoundTripper{Debug: debug}
+	}
+	return &ConformanceTestSuite{
+		Client:           c,
+		RoundTripper:     roundTripper,
+		GatewayClassName: gcName,
+		Debug:            debug,
+		Cleanup:          cleanup,
+	}
+}
+
+// Setup ensures the base resources required for conformance tests are installed
+// in the cluster. It also ensures that all relevant resources are ready.
+func (suite *ConformanceTestSuite) Setup(t *testing.T) {
+	suite.ControllerName = kubernetes.GWCMustBeAccepted(t, suite.Client, suite.GatewayClassName, 180)
```

Comment:
I tried `t.Run` first, but that didn't feel right because these items are not really tests as much as they are setup steps. (The GatewayClass status == accepted is relatively close to a test, but the rest are not). Instead I've added logging before each step, which does seem to make the output much clearer here.